pkg/config/config.go

```diff
@@ -349,6 +353,39 @@ func (dc *DeploymentConfig) listUnusedModules() map[string][]string {
 	return unusedModules
 }
 
+func (dc *DeploymentConfig) listUnusedDeploymentVariables() []string {
+	var usedVars = []string{}
+	for _, connections := range dc.moduleConnections {
+		for _, conn := range connections {
+			if conn.kind == deploymentConnection {
+				for _, v := range conn.sharedVariables {
+					if !slices.Contains(usedVars, v) {
+						usedVars = append(usedVars, v)
+					}
+
+				}
+			}
+		}
+	}
+
+	// variables that are either required or automatically constructed and
+	// applied; these should not be returned as unused otherwise no blueprints
+	// could be validated
+	requiredVars := []string{"labels", "deployment_name"}
+	var unusedVars = []string{}
+	for _, v := range maps.Keys(dc.Config.Vars) {
```

Comment:
`for v, _ := range dc.Config.Vars {`

Reply:
Addressed. This was a relic of a refactor I didn't catch. Thanks.